pkg/controller/service/servicecontroller.go

```diff
@@ -482,15 +484,18 @@ func (s *ServiceController) loadBalancerName(service *api.Service) string {
 }
 
 func getPortsForLB(service *api.Service) ([]*api.ServicePort, error) {
+	var protocol api.Protocol
+
 	ports := []*api.ServicePort{}
 	for i := range service.Spec.Ports {
-		// TODO: Support UDP. Remove the check from the API validation package once
-		// it's supported.
 		sp := &service.Spec.Ports[i]
-		if sp.Protocol != api.ProtocolTCP {
-			return nil, fmt.Errorf("load balancers for non TCP services are not currently supported.")
-		}
+		// The check on protocol was removed here.  The cloud provider itself is now responsible for all protocol validation
 		ports = append(ports, sp)
+		if protocol == "" {
+			protocol = sp.Protocol
+		} else if protocol != sp.Protocol && wantsLoadBalancer(service) {
+			return nil, fmt.Errorf("mixed protocol external load balancers are not supported.")
```

Comment:
Same as above.  If it is preferred to record an event here, let's change that throughout.  This basically replaces the previous error that was returned in the same manner that indicated TCP wasn't supported at all.